### ISSUE TITLE
Endless loop in error_handling

### ISSUE THREAD

**Issue by shu-bc** [neo4jrb/neo4j-ruby-driver#49]
I have encountered an endless loop, which sends Errbit `stack level too deep` error and detail backtrace message like this.
[backtrace.txt](https://github.com/neo4jrb/neo4j-ruby-driver/files/5148104/backtrace.txt)

2 minutes later, Errbit got this error, probably related
```
code: `f`, error: `15`, state: `4`, error_context: `secure_openssl_recv(/tmp/seabolt/src/seabolt/src/bolt/communication-secure-openssl.c:488), SSL_read error code: 5, underlying error code: 0`
```

My environment:
ruby: 2.6.5
neo4j-ruby-driver: 1.7.0
activegraph: 10.0.1
neo4j: 3.4.0

**Comment by klobuczek** [comment]
@shu-bc can you reproduce the problem. I can the endless loop. I think this can happen if the server goes down during a query or retrieving error details. Do you see other scenario? Thanks for the bug report.

**Comment by klobuczek** [comment]
Fixed in 1.7.1
